packages/node_modules/@ciscospark/spark-core/src/lib/spark-internal-core-plugin-mixin.js

```diff
@@ -51,15 +51,17 @@ export default function mixinSparkInternalCorePlugins(State, config, interceptor
       }));
     }
 
-
-    if (constructor.prototype._definition.ready) {
-      const {fn, depList} = State.prototype._derived.ready;
-      const def = {
-        deps: depList.concat(`${name}.ready`),
-        fn
-      };
-
-      createDerivedProperty(State.prototype, 'ready', def);
+    // Only mess with the plugin's derived properties if it's an amp-state plugin
+    if (constructor.prototype._definition) {
+      if (constructor.prototype._definition.ready) {
```

Comment:
Same as above.
Also there seems to be quiet a bit of duplicate code between the internal and non-internal versions of this. Is there a way to simplify this?

Reply:
Probably. When I did the internal/external split, I think I tried a few things that were all pretty terrible, but it's certainly worth investigating again. That said, I've got some idea for making the plugin system more explicit, so, that might all be worth doing in a bigger refactor.